Purchases/Error Handling/ErrorUtils.swift

```diff
@@ -257,6 +259,24 @@ extension ErrorUtils {
                      extraUserInfo: extraUserInfo)
     }
 
+    // Addes a sub-error to the userInfo of the `error` object as some extra context. Sometimes we have multiple error
+    // Conditions but only a single place to surface them. This adds the second error as extra context to help during
+    // debugging.
+    static func attach(subError: Error?, toError error: Error, extraContext: String? = nil) -> Error {
```

Comment:
might be a little cleaner as an extension?

Reply:
It is an extension 😄

Reply:
sorry i meant like a mutating method on the `toError`

Reply:
🤦‍♂️ yeah, ok, updated!